test/libwhisper/bloomFilter.cpp

```diff
@@ -111,11 +112,13 @@ BOOST_AUTO_TEST_CASE(falsePositiveRate)
 	TopicBloomFilterTest f;
 	Topic x(0xC0DEFEED); // deterministic pseudorandom value
 
-	for (int i = 1; i < 21; ++i)
+	double expectedRate = 0;
+
+	for (int i = 1; i < 50 && expectedRate < 0.5; ++i)
```

Comment:
float comparison like this is discouraged. In the olden days and in C we basically always compare the difference being above a trheshold called `EPSILON`.

Now C++11 has a [macro for that](http://www.cplusplus.com/reference/cmath/isless/). You can use `isless(expectedRate, 0.5)`